### PR TITLE
Implement file validation function

### PR DESCRIPTION
**Pull request by regular-polygon**

### CODE REVIEW

File: react-paw-mailmerge/src/components/FileUpload.tsx [hunk 1]
```diff
@@ -11,10 +11,33 @@ const FileUpload: React.FC = () => {
     setSelectedFile(file || null);
   };
 
+  const validateFile = (file: File): string[] => {
+    const errors: string[] = [];
+    const type = file.type;
+    const size_limit = 10 * 1024 * 1024;  // file.size is in bytes. 10 mb = 10 * 1024 * 1024 bytes
+
+    if (type !== 'text/csv') {
+      errors.push(`Expected file with MIME type text/csv, got ${type}.`);
```
Comment (Cat-1) [inline react-paw-mailmerge/src/components/FileUpload.tsx:20]: Good descriptive error message. Should we include the file name as well?

File: react-paw-mailmerge/src/components/FileUpload.tsx [hunk 2]
```diff
@@ -11,10 +11,33 @@ const FileUpload: React.FC = () => {
     setSelectedFile(file || null);
   };
 
+  const validateFile = (file: File): string[] => {
+    const errors: string[] = [];
+    const type = file.type;
+    const size_limit = 10 * 1024 * 1024;  // file.size is in bytes. 10 mb = 10 * 1024 * 1024 bytes
```
Comment (Cat-1) [inline react-paw-mailmerge/src/components/FileUpload.tsx:17]: Good comment, and for storing it in a reasonably named variable.

**Comment by Cat-1** [comment]
Nicely done!

**Review by Cat-1 — APPROVED** [review]
Nicely done.

File: react-paw-mailmerge/src/components/FileUpload.tsx (hunk 1 above)
Comment (regular-polygon) [reply react-paw-mailmerge/src/components/FileUpload.tsx:20]: That's a good idea. I will update this.

**Pull request merged**